tests/File/GcsFileTest.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+namespace Keboola\Test\File;
+
+use GuzzleHttp\Client;
+use Keboola\StorageApi\Exception;
+use Keboola\StorageApi\Options\FileUploadOptions;
+use Keboola\StorageApi\Options\GetFileOptions;
+use Keboola\Test\StorageApiTestCase;
+
+class GcsFileTest extends StorageApiTestCase
+{
+    /**
+     * @dataProvider uploadData
+     */
+    public function testFileUpload(string $filePath, FileUploadOptions $options): void
+    {
+        $fileId = $this->_client->uploadFile($filePath, $options);
+        $file = $this->_client->getFile($fileId, (new GetFileOptions())->setFederationToken(true));
+
+        //gcs storage is always encrypted and private. Request params 'isEncrypted' and 'isPublic' is ignored
+        $this->assertFalse($file['isPublic']);
+        $this->assertTrue($file['isEncrypted']);
+
+        $this->assertEquals(basename($filePath), $file['name']);
+        $this->assertEquals(filesize($filePath), $file['sizeBytes']);
+        $this->assertArrayHasKey('gcsCredentials', $file);
+        $this->assertArrayHasKey('gcsPath', $file);
+        sleep(1); // tests sometimes return 403, but file can be downloaded just a few seconds later
+        $this->assertEquals(file_get_contents($filePath), file_get_contents($file['url']));
+
+        $tags = $options->getTags();
+        sort($tags);
+        $fileTags = $file['tags'];
+        sort($fileTags);
+        $this->assertEquals($tags, $fileTags);
+
+        $info = $this->_client->verifyToken();
+        $this->assertEquals($file['creatorToken']['id'], (int) $info['id']);
+        $this->assertEquals($file['creatorToken']['description'], $info['description']);
+
+        if ($options->getIsPermanent()) {
+            $this->assertNull($file['maxAgeDays']);
+        } else {
+            $this->assertIsInt($file['maxAgeDays']);
+            $this->assertEquals(StorageApiTestCase::FILE_LONG_TERM_EXPIRATION_IN_DAYS, $file['maxAgeDays']);
+        }
+
+        // check attachment, download
+        $client = new Client();
+        $client->get($file['url']);
+    }
+
+    public function uploadData(): array
+    {
+        $path = __DIR__ . '/../_data/files.upload.txt';
+        $largeFilePath = sys_get_temp_dir() . '/large_abs_upload.txt';
+        $this->generateFile($largeFilePath, 16);
+
+        return [
+            [
+                $path,
+                (new FileUploadOptions())->setIsPublic(true),
```

Comment:
hmm, does this test anything?

Reply:
yeah that test, if I set public true file will be still isPublic = false